src/megachatapi_impl.cpp

```diff
@@ -1885,6 +1885,51 @@ MegaChatListItemList *MegaChatApiImpl::getChatListItems()
     return items;
 }
 
+MegaChatListItemList *MegaChatApiImpl::getChatListItemsByPeers(megachat::MegaChatPeerList *peers)
+{
+    MegaChatListItemListPrivate *items = new MegaChatListItemListPrivate();
+
+    sdkMutex.lock();
+
+    if (mClient && !terminating)
+        {
+            ChatRoomList::iterator it;
+            for (it = mClient->chats->begin(); it != mClient->chats->end(); it++)
+            {
+                bool sameParticipants = true;
+                MegaChatRoomPrivate *chat = new MegaChatRoomPrivate(*it->second);
```

Comment:
Instead of creating a new `MegaChatRoomPrivate` just for comparison of the peer list, it would be better to directly use the `it->second` for that purpose. And create the high-level (and kind of more heavy) object only if it's one of the chatrooms to be returned by this function (aka. in the block of `if (sameParticipants == true)`